packages/next-codemod/bin/upgrade.ts

```diff
@@ -30,9 +31,20 @@ async function loadHighestNPMVersionMatching(query: string) {
 }
 
 export async function runUpgrade(
-  revision: string | undefined,
+  revision: string,
   options: { verbose: boolean }
 ): Promise<void> {
+  console.log(chalk.bold(`Next.js upgrade codemode (version: ${revision})`))
+
+  if (
+    !['rc', 'latest', 'canary'].includes(revision) &&
+    !semver.valid(revision)
```

Comment:
We don't need to install `semver` we have `validate` from `compare-versions`.
x-ref: https://www.npmjs.com/package/compare-versions